### PR TITLE
ref(onboarding): Add loader script toggle to the URL

### LINKED ISSUE #76659: Add back navigation from npm / yarn to loader scipt
When selecting javascript as a platform the default docs describe how to add the loader script.
The user can select to use yarn / npm instead but there is no going back from that.

We should also persist this state in the URL.

### PR DESCRIPTION
**Pull request by priscilawebdev**
This PR improves the code responsible for displaying the Loader Script in the onboarding/project creation flow. Instead of relying on a component's internal state to hide the loader, we now use a URL parameter showLoader (of type boolean) to control its visibility. Additionally, if users opt to follow the npm/yarn installation instructions, a similar alert will be shown, providing them with the option to return to the Loader Script instructions.

https://github.com/user-attachments/assets/d257bcae-cc72-4a42-b124-350b5555e071

requires https://github.com/getsentry/getsentry/pull/14976

closes https://github.com/getsentry/sentry/issues/76659

**Note**

the content is currently not loading smoothly, but I will be addressing this issue in a follow-up

### CODE REVIEW

File: static/app/components/onboarding/productSelection.tsx [hunk 1]
```diff
@@ -466,17 +492,34 @@ export function ProductSelection({
...
+                goHere: (
+                  <SkipLazyLoaderButton
+                    onClick={handleToggleLoader}
+                    size="xs"
+                    priority="default"
+                  >
+                    {t('View loader instructions')}
+                  </SkipLazyLoaderButton>
+                ),
+              })}
```
[the remark this answers is not in the document]
Comment (ArthurKnaus) [reply static/app/components/onboarding/productSelection.tsx:522]: can be a follow-up

[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/onboarding/productSelection.tsx:522]: you are right and I plan to do that in a follow-up. The Product Selection component has so many other things, for instance the npm install message and I want to remove that as well... so if you do not have strong feelings, I would like to refactor this component in a separate PR completely

File: static/app/components/onboarding/productSelection.tsx [hunk 2]
```diff
@@ -384,6 +392,24 @@ export function ProductSelection({
...
+    }
+
+    router.replace({
+      pathname: router.location.pathname,
+      query: {
+        ...router.location.query,
+        showLoader: !showLoader,
+      },
+    });
+  }, [router, showLoader, platform, projectId, organization]);
```
Comment (ArthurKnaus) [inline static/app/components/onboarding/productSelection.tsx:411]: You could put this + accessing the query param into a hook which can be used here and in `platform.tsx`:
```ts
const [
  showLoaderOnboarding,
  toggleLoaderOnboarding
] = useLoaderOnboardingState()
```
Then it is ensured that both components access the same query parameter.

**Review by ArthurKnaus — APPROVED** [review]

File: static/app/components/onboarding/productSelection.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/components/onboarding/productSelection.tsx:411]: that is a very good idea! will implement that 🙏

**Pull request merged**